api/v1alpha1/orchestrator_types.go

```diff
@@ -20,149 +20,182 @@ import (
 }
 
 type NotificationConfig struct {
-	Enabled   bool   `json:"enabled,omitempty"`
-	Port      int    `json:"port,omitempty"`
-	Sender    string `json:"sender,omitempty"`
-	Recipient string `json:"replyTo,omitempty"`
-}
+	// Determines whether to install the Notifications Email plugin
+	// Requires setting the hostname and credentials in backstage secret
+	// The secret backstage-backend-auth-secret is created as pre-requisite
+	// See plugin configuration at https://github.com/backstage/backstage/blob/master/plugins/notifications-backend-module-email/config.d.ts
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled,omitempty"`
 
-type RHDHPlugins struct {
-	NpmRegistry         string                   `json:"npmRegistry,omitempty"`
-	Scope               string                   `json:"scope,omitempty"`
-	Plugins             map[string]PluginDetails `json:"plugins,omitempty"`
-	NotificationsConfig NotificationConfig       `json:"notificationsConfig,omitempty"`
+	// SMTP server port
+	// +kubebuilder:default=587
+	Port int `json:"port,omitempty"`
+
+	// Email address of the Sender
+	// +kubebuilder:default=""
+	Sender string `json:"sender,omitempty"`
+
+	// Email address of the Recipient
+	// +kubebuilder:default=""
+	Recipient string `json:"replyTo,omitempty"`
 }
 
 type Postgres struct {
-	ServiceName      string             `json:"serviceName,omitempty"`
-	ServiceNameSpace string             `json:"serviceNamespace,omitempty"`
-	AuthSecret       PostgresAuthSecret `json:"authSecret,omitempty"`
-	DatabaseName     string             `json:"database,omitempty"`
+	// Name of the Postgres DB service to be used by platform services
+	// +kubebuilder:validation:Required
+	ServiceName string `json:"serviceName"`
+
+	// Namespace of the Postgres DB service to be used by platform services
+	// +kubebuilder:validation:Required
+	ServiceNameSpace string `json:"serviceNamespace"`
+
+	// PostgreSQL connection credentials details
+	// +kubebuilder:validation:Required
+	AuthSecret PostgresAuthSecret `json:"authSecret"`
+
+	// Existing database instance used by data index and job service
+	// +kubebuilder:validation:Required
+	DatabaseName string `json:"database"`
 }
 
 type PostgresAuthSecret struct {
-	SecretName  string `json:"name,omitempty"`
-	UserKey     string `json:"userKey,omitempty"`
-	PasswordKey string `json:"passwordKey,omitempty"`
+	// Name of existing secret to use for PostgreSQL credentials.
+	// +kubebuilder:validation:Required
+	SecretName string `json:"name"`
+
+	// Name of key in existing secret to use for PostgreSQL credentials.
+	// +kubebuilder:validation:Required
+	UserKey string `json:"userKey"`
+
+	// Name of key in existing secret to use for PostgreSQL credentials.
+	// +kubebuilder:validation:Required
+	PasswordKey string `json:"passwordKey"`
 }
 
-type OrchestratorPlatform struct {
-	Namespace          string             `json:"namespace,omitempty"`
+type ServerlessWorkflow struct {
```

Comment:
why not `ServerlessWorkflowConfig` as for RHDH and notification?

Reply:
Still waiting on confirmation, but we will most likely be eliminating the `OrchestratorPlatform` struct entirely. it is related to the confusing `orchestrator` field

Reply:
Updated field to platform